src/lib.rs

```diff
@@ -158,21 +158,177 @@ mod core {
 use core::fmt;
 
 #[cfg(not(feature = "std"))]
+#[macro_use]
 extern crate alloc;
+
 #[cfg(not(feature = "std"))]
 use alloc::Vec;
+#[cfg(not(feature = "std"))]
+use alloc::string::String;
+#[cfg(not(feature = "std"))]
+use alloc::borrow::ToOwned;
+#[macro_use]
+extern crate serde_derive;
+
+extern crate serde;
+
+use serde::ser::{SerializeStruct, SerializeSeq, Serialize, Serializer};
+
+/// Types allowed to be value in the context vector
+#[derive(Debug, Clone)]
+pub enum ThrowContextValues {
+    ///Boolean
+    Bool(bool),
+    ///Int8
+    Int8(i8),
```

Comment:
One open question I have here is whether we want to support all numbers independently, or just the biggest ones (i64, u64, f64).

Supporting fewer numbers would mean a simpler context value type, but it would have the disadvantage of having less information to pass to the `serde::Serializer`. I'm not sure which would be the best idea, but we can err on the current side.

Reply:
while json only supports f64, I was thinking about another serialization formats,  that support small numbers,  it will be more efficient to communicate that that using serde.